### PR TITLE
Upgrade to be an importable Python package.

### PR DESCRIPTION
**Pull request by bjlittle**
This PR addresses a pain-point for all iris users ... hooking in the iris-sample-data to the `iris/etc/site.cfg` so that it can be used by `iris.sample_data_path` function.

At present, this is a **manual** task where *everyone* has to ...
* understand what is going wrong i.e. "why doesn't `iris.sample_data_path` just work?"
* figure out how to fix it
* rinse-and-repeat for all their iris installations

Life is way too short.

This PR packages up the iris-sample-data into an importable package that can easily be installed. 
* This means that iris can detect it and use it (separate iris PR on the way)
* Allows iris-sample-data to be packaged into a conda recipe :smile:

### CODE REVIEW

**Comment by bjlittle** [comment]
ping @marqh @dkillick @pelson

**Comment by marqh** [comment]
looks great to me, is there a companion pr on iris to hook this in?

**Comment by marqh** [comment]
if there are no further comments, I am minded to merge this

going once .....

**Comment by bjlittle** [comment]
@marqh working on the iris PR now :wink: 

Do you wan to wait of that, so that you can test them together?

**Comment by marqh** [comment]
> Do you wan to wait of that, so that you can test them together?

i don't think that is required

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    version=extract_version(),
+    description=DESCRIPTION,
+    long_description=extract_description(),
+    platforms=['Linux', 'Max OS X', 'Windows'],
+    license='Open Government',
+    url='https://github.com/SciTools/iris-sample-data',
+    packages=[NAME],
+    package_data=extract_package_data(),
+    classifiers=[
+        'License :: OSI Approved :: Open Government License',
```
Comment (QuLogic) [inline setup.py]: This is not a classifier [listed on PyPI](https://pypi.python.org/pypi?%3Aaction=list_classifiers).

Comment (QuLogic) [inline setup.py]: Also, I don't see it [listed on OSI](https://opensource.org/licenses/alphabetical).

Comment (bjlittle) [inline setup.py]: Thanks, I'll consult as to what a [Open Government](http://www.nationalarchives.gov.uk/doc/open-government-licence/version/2/) licence actually is ...

**Pull request merged**